pytest_deep_analysis/runtime/plugin.py

```
@@ -0,0 +1,295 @@
+    """Hook called before test execution."""
+    if _plugin_instance and _plugin_instance.enabled:
+        _plugin_instance.start_test_execution(item)
+
+
+def pytest_runtest_call(item: Item):
+    """Hook called during test execution."""
+    # Execution tracing happens automatically via sys.settrace
+    pass
+
+
+def pytest_runtest_teardown(item: Item, nextitem):
```

Ruff (0.14.4)

Unused function argument: `nextitem`

(ARG001)